### PR TITLE
feat: added postModal and posts sorting functionality

### PR DESCRIPTION
**Pull request by Hardiegogo**
- Added PostModal
- Added useOutSideClick custom hook 
- Added sorting functionality for feed posts

Screenshots:
image
image

### CODE REVIEW

File: src/components/PostModal.jsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    }
+    useOutsideClick(refr,()=>dispatch(postModalOff()))
+    return (
+      <div className="fixed min-h-screen min-w-[100vw] top-0 left-0 grid place-items-center bg-[rgba(0,0,0,0.5)] z-20" >
+        <div ref={refr}className="border-solid border-2 border-black bg-lightOrange p-6 rounded shadow-neu w-fit min-w-[30rem]  h-fit">
+        <div className="flex gap-2">
+          <img
+            className="w-12 rounded-full object-cover h-12 "
+            src={user.userProfile}
+            alt=""
```
Comment (nerdyspook) [inline src/components/PostModal.jsx:44]: You can give the alt text as profile pic here, as this image contains the profile pic of the user right?

**Review by nerdyspook — APPROVED** [review]
Code structure looks well organized. And the UI looks great.

**Review by AmishMishra11 — APPROVED** [review]
Great work on the ui
Code is clean 
you can merge

File: src/components/PostModal.jsx (hunk 1 above)
Comment (Hardiegogo) [reply src/components/PostModal.jsx:44]: Yes

**Pull request merged**